src/rgw/rgw_lua_s3.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <string>
+#include "include/common_fwd.h"
+
+class req_state;
+class RGWREST;
+class OpsLogSocket;
+namespace rgw::sal {
+  class RGWRadosStore;
+}
+
+namespace rgw::lua::s3 {
+
+// execute a lua script in the S3Request context
+int execute(
```

Comment:
where is this function defined? i have
```
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-4580-g89aeb16/rpm/el8/BUILD/ceph-16.0.0-4580-g89aeb16/build/lib/librgw.so: undefined reference to `rgw::lua::s3::execute(rgw::sal::RGWRadosStore*, RGWREST*, OpsLogSocket*, req_state*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-4580-g89aeb16/rpm/el8/BUILD/ceph-16.0.0-4580-g89aeb16/build/lib/librgw.so: undefined reference to `rgw::lua::read_script(rgw::sal::RGWRadosStore*, rgw_user const&, optional_yield, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&)'
collect2: error: ld returned 1 exit status
```

when linking `librgw.so` . see https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=centos8,DIST=centos8,MACHINE_SIZE=gigantic/44832//consoleFull

Reply:
@tchaikov fixed